exercises/1-User_Mode/challenge_101/test/test_101_challenge.py

```diff
@@ -0,0 +1,131 @@
+    # -or-
+    python3 -m unittest -v -k boundary_01  # The one test cases that should *always* pass
+    # -or-
+    python3 -m unittest -v -k normal_01    # If you can make this pass, you're doing well
+"""
+
+# Standard Imports
+# Third Party Imports
+from test.challenge_101_test_class import Challenge101Test, SignalResults
+from hobo.subprocess_wrapper import execute_subprocess_cmd
+# Local Imports
+
+
+class Challenge101Normal(Challenge101Test):
+    """Defines normal test cases."""
+
+    def test_normal_01(self):
+        """Start, SIGQUIT, then SIGKILL.
+
+        Test class will implicitly add a SIGKILL.
+        """
+        self.check_signals([SignalResults(3, False)])
+        self.expect_stderr(['Ignoring signal: 3'])
+        self.run_test()
+
+    def test_normal_02(self):
+        """Start, SIGINT, then SIGKILL.
+
+        Explicitly added SIGKILL."""
+        self.check_signals([SignalResults(2, False), SignalResults(9, True)])
+        self.expect_stderr(['Ignoring signal: 2'])
+        self.run_test()
+
+
+class Challenge101Boundary(Challenge101Test):
+    """Defines boundary test cases."""
+
+    def test_boundary_01(self):
+        """One signal: Start, then SIGKILL."""
+        self.check_signals([SignalResults(9, True)])
+        self.verify_stderr_empty()  # No handled signals, no output
+        self.run_test()
+
+    def test_boundary_02(self):
+        """11 signals: 10 non-handled signals, then SIGKILL."""
+        # LOCAL VARIABLES
+        sig_res_list = []  # List of SignalResults for this test case
+        exp_err_list = []  # List of expected stderr output
+
+        # SETUP INPUT
+        # Handled signals
+        for sig_num in range(20, 31):
+            sig_res_list.append(SignalResults(sig_num, False))
+            exp_err_list.append(f'Ignoring signal: {sig_num}')
+        # Non-handled signals
+        sig_res_list.append(SignalResults(9, True))
+
+        # RUN IT
+        self.check_signals(sig_res_list)
+        self.expect_stderr(exp_err_list)
+        self.run_test()
+
+    def test_boundary_03(self):
+        """Smallest valid signal: SIGHUP."""
+        self.check_signals([SignalResults(1, False)])
+        self.expect_stderr(['Ignoring signal: 1'])
+        self.run_test()
+
+    def test_boundary_04(self):
+        """Largest valid signal: SIGRTMAX."""
+        self.check_signals([SignalResults(64, False)])
+        self.expect_stderr(['Ignoring signal: 64'])
+        self.run_test()
+
+
+class Challenge101Special(Challenge101Test):
+    """Defines special test cases."""
+
+    def test_special_01(self):
+        """Start, SIGSTOP, SIGCONT, then SIGKILL.
+
+        SIGSTOP pauses the process.  SIGCONT wakes it back up.  SIGKILL kills it.
+        """
+        self.check_signals([SignalResults(19, False), SignalResults(18, False),
+                            SignalResults(9, True)])
+        self.expect_stderr(['Ignoring signal: 18'])
```

Comment:
Change expected output to just be 'Ignoring signal' since printf() is not a viable solution for signal handlers.

Reply:
Updated.